### ISSUE TITLE
dts-generator should handle optional flag for interface methods

### ISSUE THREAD

**Issue by codeworrior** [SAP/ui5-typescript#43]
Since a few releases, UI5's `api.json` files can contain an optional flag for interface methods.

The dts-generator should read this flag and generate interface methods that have `optional:true` with a trailing question mark (the `TypeScript` notation for an optional interface method):

An `api.json` of 
```json
	{
		"kind": "interface",
		"name": "sap.ui.core.IFormContent",
		"methods": [
			{
				"name": "getFormDoNotAdjustWidth",
				"visibility": "public",
				"optional": true,
				"returnValue": {
					"type": "boolean",
				}
			}
		]
	}
```

should result in a method like

```ts
interface IFormContent {
  getFormDoNotAdjustWidth? ():boolean
}
```

**Comment by bd82** [comment]
Released in 1.60.5
